src/telemetry.ts

```diff
@@ -751,6 +767,10 @@ export class IEventNamePropertyMapping {
          * Mimetype of the output.
          */
         mimeType: string;
+        /**
+         * Whether the package was detected in an existing file (upon open, upon save, upon close) or when it was being used during execution.
+         */
+        when: 'onExecution' | 'onOpenCloseOrSave';
```

Comment:
Helps determine whether the vega mime types & similar are generated as part of execution or whether they are merely detected in existing notebooks.